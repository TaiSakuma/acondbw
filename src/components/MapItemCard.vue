<template>
  <div class="mapitemcard">
    <div v-if="$apollo.queries.map.loading">loading...</div>
    <div v-else-if="error">Error: cannot load data</div>
    <v-card v-else-if="map" outlined hover style="max-width: 980px;">
      <div @click="$emit('expand')" style="cursor: default;">
        <v-container fluid class="pa-0">
          <v-row class="ma-0 px-0">
            <v-col cols="12" md="4" class="pa-2">
              <div class="caption grey--text">Name</div>
              <div class="font-weight-medium primary--text">
                <span @click.stop>
                  <router-link :to="'/maps/item/' + map.name" v-text="map.name"></router-link>
                </span>
              </div>
            </v-col>
            <v-col cols="6" md="4" class="pa-2">
              <div class="caption grey--text">Date posted</div>
              <div v-text="map.datePosted"></div>
            </v-col>
            <v-col cols="5" md="3" class="pa-2">
              <div class="caption grey--text">Mapper</div>
              <div v-text="map.mapper"></div>
            </v-col>
            <v-col cols="1" md="1" align-self="end" class="pa-2">
              <div v-if="collapsible">
                <v-row align="start" justify="end" class="px-1">
                  <v-tooltip bottom open-delay="800">
                    <template v-slot:activator="{ on }">
                      <v-btn
                        icon
                        @click.stop="
                      collapsed ? $emit('expand') : $emit('collapse')
                    "
                        v-on="on"
                      >
                        <v-icon>
                          {{
                          collapsed ? "mdi-chevron-down" : "mdi-chevron-up"
                          }}
                        </v-icon>
                      </v-btn>
                    </template>
                    <span>{{ collapsed ? "Expand" : "Collapse" }}</span>
                  </v-tooltip>
                </v-row>
              </div>
            </v-col>
          </v-row>
          <v-expand-transition>
            <v-row class="mx-0 mb-3 px-0 collapsible" v-show="!(collapsible && collapsed)">
              <v-col cols="12" md="8" offset-md="4" class="pa-2">
                <div class="caption grey--text">Paths</div>
                <ul v-if="map.mapFilePaths">
                  <li
                    v-for="(edgep, index) in map.mapFilePaths.edges"
                    :key="index"
                    v-text="edgep.node.path"
                  ></li>
                </ul>
              </v-col>
              <v-col cols="12" md="8" offset-md="4" class="pa-2">
                <div class="caption grey--text">Note</div>
                <div>
                  <ul v-if="map.note">
                    <li
                      v-for="(line, index) in map.note.split('\n')"
                      :key="index"
                    >{{ line.replace(/^- */, "") }}</li>
                  </ul>
                </div>
              </v-col>
              <v-col cols="12" md="8" offset-md="4" class="pa-2">
                <div class="caption grey--text">Beams</div>
                <ul v-if="map.beams">
                  <li v-for="(edgep, index) in map.beams.edges" :key="index">
                    <router-link :to="'/beams/item/' + edgep.node.name" v-text="edgep.node.name"></router-link>
                  </li>
                </ul>
              </v-col>
            </v-row>
          </v-expand-transition>
        </v-container>
      </div>
    </v-card>
    <div v-else>Nothing to show here.</div>
  </div>
</template>

<script>
import MAP from "@/graphql/Map.gql";

export default {
  name: "MapItemCard",
  props: {
    mapName: { default: null },
    collapsed: { default: false },
    collapsible: { default: false }
  },
  data() {
    return {
      map: null,
      error: null
    };
  },
  apollo: {
    map: {
      query: MAP,
      variables() {
        return {
          name: this.mapName
        };
      },
      result(result) {
        this.error = null;
        if (result.error) {
          this.error = true;
        }
      }
    }
  }
};
</script>
