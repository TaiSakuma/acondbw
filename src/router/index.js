import Vue from "vue";
import VueRouter from "vue-router";
import Home from "@/views/Home.vue";
import ProductTop from "@/views/ProductTop.vue";

import ALL_MAPS from "@/graphql/AllMaps.gql";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: Home,
  },
  {
    path: "/about",
    name: "about",
    component: () =>
      import(/* webpackChunkName: "about" */ "@/views/About.vue"),
  },
  {
    path: "/simulations",
    component: ProductTop,
    props: {
      title: "Simulations",
      icon: "mdi-creation",
      itemPageName: "SimulationItem",
    },
    children: [
      {
        path: "",
        name: "SimulationList",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationList.vue"
          ),
      },
      {
        path: "item/:name",
        name: "SimulationItem",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationItem.vue"
          ),
      },
    ],
  },
  {
    path: "/maps",
    component: ProductTop,
    props: { title: "Maps", icon: "map", itemPageName: "MapItem" },
    children: [
      {
        path: "",
        name: "MapList",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapList.vue"),
        props: {
          productTypeName: { singular: "map", plural: "maps" },
          query: ALL_MAPS,
          queryName: "allMaps",
          productIdFieldName: "mapId",
          productItemCard: "MapItemCard",
          productAddForm: "MapAddForm",
          disableAdd: process.env.VUE_APP_ACONDBW_MAP_MUTATION_DIALOG != "true",
        },
      },
      {
        path: "item/:name",
        name: "MapItem",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapItem.vue"),
      },
    ],
  },
  {
    path: "/beams",
    component: ProductTop,
    props: {
      title: "Beams",
      icon: "mdi-spotlight-beam",
      itemPageName: "BeamItem",
    },
    children: [
      {
        path: "",
        name: "BeamList",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamList.vue"),
      },
      {
        path: "item/:name",
        name: "BeamItem",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamItem.vue"),
      },
    ],
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

export default router;
