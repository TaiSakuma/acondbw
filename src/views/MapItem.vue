<template>
  <div class="mapitem">
    <v-container fluid>
      <v-row class="display-1 mx-1 mt-3 primary--text" style="max-width: 980px;">
        <v-col col="8" class="pa-0 ma-0">
          <span class="me-2">
            <v-icon>map</v-icon>
          </span>Maps
          <v-icon large color="primary">mdi-chevron-right</v-icon>
          {{ $route.params.name }}
        </v-col>
      </v-row>
      <div class="map-item" style="position: relative;">
        <DevToolLoadingStateOverridingMenu @state="devtoolState = $event"></DevToolLoadingStateOverridingMenu>
        <div v-if="state == State.LOADING" class="mx-2 pt-5">
          <v-progress-circular indeterminate :size="26" color="grey"></v-progress-circular>
        </div>
        <div v-else-if="state == State.ERROR" class="mx-2 pt-5">
          <v-card outlined style="max-width: 980px;">
            <v-card-text>Error: cannot load data</v-card-text>
          </v-card>
        </div>
        <div v-else-if="state == State.LOADED || state == State.EMPTY">
          <v-container fluid class="pa-0">
            <v-row
              align="start"
              justify="end"
              class="ma-0 px-0 pt-3 pb-1"
              style="max-width: 980px;"
            >
              <v-tooltip bottom open-delay="800">
                <template v-slot:activator="{ on }">
                  <v-btn text icon exact to="/maps" v-on="on">
                    <v-icon>mdi-arrow-left</v-icon>
                  </v-btn>
                </template>
                <span>Back to Maps</span>
              </v-tooltip>
              <v-tooltip bottom open-delay="800">
                <template v-slot:activator="{ on }">
                  <v-btn icon @click="$apollo.queries.map.refetch();" v-on="on">
                    <v-icon>mdi-refresh</v-icon>
                  </v-btn>
                </template>
                <span>Refresh</span>
              </v-tooltip>
              <v-spacer></v-spacer>
            </v-row>
          </v-container>
          <div v-if="state == State.LOADED">
            <MapItemCard
              :mapId="map.mapId"
              :collapsible="false"
              v-on:deleted="$router.push('/maps')"
            ></MapItemCard>
          </div>
        </div>
        <div v-else class="mx-2 pt-5">
          <v-card outlined style="max-width: 980px;">
            <v-card-text>Nothing to show here.</v-card-text>
          </v-card>
        </div>
      </div>
    </v-container>
  </div>
</template>
      
      

<script>
import gql from "graphql-tag";
import MAP_FRAGMENT from "@/graphql/MapFragment.gql";

import MapItemCard from "@/components/MapItemCard";

import DevToolLoadingStateOverridingMenu from "@/components/DevToolLoadingStateOverridingMenu";

const State = {
  LOADED: 0,
  EMPTY: 1,
  LOADING: 2,
  ERROR: 3,
  NONE: 4
};

const GqlMapName = gql`
  query MapIdName($name: String) {
    map(name: $name) {
      ...map
    }
  }
  ${MAP_FRAGMENT}
`;

export default {
  name: "mapItem",
  components: {
    MapItemCard,
    DevToolLoadingStateOverridingMenu
  },
  data() {
    return {
      map: null,
      error: null,
      devtoolState: null,
      State: State
    };
  },
  computed: {
    state() {
      if (this.devtoolState) {
        return this.devtoolState;
      }

      if (this.loading) {
        return State.LOADING;
      } else if (this.error) {
        return State.ERROR;
      } else if (this.map) {
        return State.LOADED;
      } else {
        return State.NONE;
      }
    },
    loading() {
      return this.$apollo.queries.map.loading;
    }
  },
  apollo: {
    map: {
      query: GqlMapName,
      variables() {
        return {
          name: this.$route.params.name
        };
      },
      result(result) {
        this.error = null;
        if (result.error) {
          this.error = true;
        }
      }
    }
  }
};
</script>
