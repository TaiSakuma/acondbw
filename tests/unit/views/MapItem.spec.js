import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { mount, createLocalVue } from "@vue/test-utils";

import MapItem from "@/views/MapItem.vue";
import MapItemCard from "@/components/MapItemCard.vue";

import router from "@/router";

Vue.use(Vuetify);
Vue.use(VueRouter);

describe("MapItem.vue", () => {
  let localVue;

  function createWrapper(loading = false) {
    return mount(MapItem, {
      localVue,
      router,
      mocks: {
        $apollo: {
          queries: {
            map: {
              loading: loading
            }
          }
        }
      },
      stubs: {
        MapItemCard: true
      }
    });
  }

  beforeEach(function() {
    localVue = createLocalVue();
  });

  it("match snapshot", async () => {
    const wrapper = createWrapper();
    wrapper.setData({
      map: {
        mapId: "1013",
        name: "lat20200201",
      }
    });
    await Vue.nextTick();
    expect(wrapper.find(MapItemCard).props().mapName).toBe("lat20200201");
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("loading", async () => {
    const loading = true;
    const wrapper = createWrapper(loading);
    await Vue.nextTick();
    expect(wrapper.text()).toContain('loading')
  });

  it("error", async () => {
    const wrapper = createWrapper();
    wrapper.setData({
      error: true
    });
    await Vue.nextTick();
    expect(wrapper.text()).toContain('Error: cannot load data')
  });

  it("none", async () => {
    const wrapper = createWrapper();
    await Vue.nextTick();
    expect(wrapper.text()).toContain('Nothing to show here')
  });
  
});
