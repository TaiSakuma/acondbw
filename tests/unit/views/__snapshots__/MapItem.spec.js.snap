// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapItem.vue match snapshot 1`] = `
<div class="mapitem">
  <div class="container container--fluid">
    <div class="row display-1 mx-1 mt-3 primary--text" style="max-width: 980px;">
      <div col="8" class="pa-0 ma-0 col"><span class="me-2"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">map</i></span>Maps
        <i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light primary--text" style="font-size: 36px;"></i>

      </div>
    </div>
    <div class="map-item" style="position: relative;"><button type="button" class="v-btn v-btn--absolute v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" style="top: -15px; right: -10px;"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-nut theme--light grey--text text--lighten-1" style="font-size: 12px;"></i></span></button>
      <div class="v-menu">
        <!---->
      </div>
      <div>
        <div class="container pa-0 container--fluid">
          <div class="row ma-0 px-0 pt-3 pb-1 align-start justify-end" style="max-width: 980px;"><a href="/maps" class="v-btn v-btn--flat v-btn--icon v-btn--round v-btn--router v-btn--text theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-arrow-left theme--light"></i></span></a><span class="v-tooltip v-tooltip--bottom"><!----></span> <span class="v-tooltip v-tooltip--bottom"><!----></span><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-refresh theme--light"></i></span></button>
            <div class="spacer"></div>
          </div>
        </div>
        <div>
          <mapitemcard-stub mapid="1013"></mapitemcard-stub>
        </div>
      </div>
    </div>
  </div>
</div>
`;
